app/models/projects/acts_as_customizable_patches.rb

```diff
@@ -0,0 +1,248 @@
+        .where(id: custom_field_values.pluck(:custom_field_id))
+        .pluck(:id, :custom_field_section_id)
+        .to_h
+    end
+
+    def validate_custom_values
+      # overrides acts_as_customizable
+      # validate custom values only of a specified section
+      # instead of validating ALL custom values like done in acts_as_customizable
+      set_default_values! if new_record?
+
+      custom_field_values
+        .select { |custom_value| of_specified_custom_field_section?(custom_value) }
+        .reject(&:marked_for_destruction?)
+        .select(&:invalid?)
+        .each { |custom_value| add_custom_value_errors! custom_value }
+    end
+
+    def of_specified_custom_field_section?(custom_value)
+      if _limit_custom_fields_validation_to_section_id.present?
+        custom_field_section_ids[custom_value.custom_field_id] == _limit_custom_fields_validation_to_section_id
+      else
+        true # validate all custom values if no specific section was specified
+      end
+    end
+
+    def custom_field_values=(values)
+      # overrides acts_as_customizable
+      # we need to query the available custom fields on a global level when updating custom field values
+      # in order to support implicit activation of custom fields when values are provided during an update
+      self._query_available_custom_fields_on_global_level = true # set to false in after_save hook
+      set_custom_field_values_method_from_acts_as_customizable_module(values)
+    end
+
+    # we cannot call super as the code in acts_as_customizable is shipped in a module
+    # thus copy and pasted the code from acts_as_customizable here
+    def set_custom_field_values_method_from_acts_as_customizable_module(values)
+      return unless values.is_a?(Hash) && values.any?
+
+      values.with_indifferent_access.each do |custom_field_id, val|
+        existing_cv_by_value = custom_values_for_custom_field(id: custom_field_id)
+                                 .group_by(&:value)
+                                 .transform_values(&:first)
+        new_values = Array(val).map { |v| v.respond_to?(:id) ? v.id.to_s : v.to_s }
+
+        if existing_cv_by_value.any?
+          assign_new_values custom_field_id, existing_cv_by_value, new_values
+          delete_obsolete_custom_values existing_cv_by_value, new_values
+          handle_minimum_custom_value custom_field_id, existing_cv_by_value, new_values
+        end
+      end
+    end
+
+    # overrides acts_as_customizable
+    # we need to query the available custom fields on a global level when
+    # trying to set a custom field which is not enabled via e.g. custom_field_123="foo"
+    def for_custom_field_accessor(method_symbol)
+      match = /\Acustom_field_(?<id>\d+)=?\z/.match(method_symbol.to_s)
+      if match
+        custom_field = available_custom_fields(global: true).find { |cf| cf.id.to_s == match[:id] }
+        if custom_field
+          yield custom_field
+        end
+      end
+    end
```

Comment:
This method also can be simplified by using the `with_all_available_custom_fields` and calling `super`.

```suggestion
    def for_custom_field_accessor(method_symbol)
      with_all_available_custom_fields { super }
    end
```